addons/web_editor/static/src/js/wysiwyg/widgets/link_popover_widget.js

```diff
@@ -85,6 +85,7 @@ export class LinkPopoverWidget {
             tooltips.push(Tooltip.getOrCreateInstance(el));
         }
         let popoverShown = true;
+        const mailWrapper = this.wysiwyg.odooEditor.editable.querySelector('.o_mail_wrapper');
```

Comment:
Can't we just set the `editable` as boundary?

Reply:
@deso-odoo Yes, we can use it, but in this case, the link popover will appear right next to the sidebar with no gap between them, making it seem as though part of the popover is hidden behind the sidebar. To prevent this, I used mail_wrapper. however, editable also works fine.

Reply:
@dhba-odoo If the screnshot you are showing is the situation using `editable`, then I think it is perfectly acceptable. I don't think it looks like part of the popover is hidden because of the rounded corners which makes it obvious that what you are seeing is indeed the end of the popover. Therefore, I think it would be better to use `editable` than to look for `.o_mail_wrapper`, which is specific to `mass_mailing` inside`link_popover_widget.js` which is more generic and should not be aware that `mass_mailing` even exists.

Reply:
@dmo-odoo applied the requested changes.

Reply:
@dhba-odoo Thank you! However, I realized my suggestion would have consequences outside of Email Marketing, so I asked Lionel (lebl) about it and he didn't like the changes. I had a look and actually `viewport` should work since the mail wrapper and the editable and stuff are all inside an iframe while the sidebar (snippets menu) is outside of it. In fact, it seems the issue occurs because the iframe has a padding to account for the sidebar instead of a margin. Could you try by changing it to margin to see if it solves the issue? Thanks!

Reply:
I think that in 17.0 the mail wrapper, editable and sidebar are all within the same iframe.

Reply:
You are right but I think the margin stuff can work even if they are all inside the iframe. So the idea is to remove the padding-right that is on the body of the iframe and replace it by a margin-right on the div that has the id `iframe_target`. Let me know if it solves the issue.

Reply:
I tried the you have describe, but it didn't work.

Reply:
Yes, you're right. I tried to fix another issue, but it didn't fix the original one, sorry about that.

What about this?
```diff
diff --git a/addons/mass_mailing/static/src/scss/mass_mailing.wysiwyg.scss b/addons/mass_mailing/static/src/scss/mass_mailing.wysiwyg.scss
index 27e5e5433a86..85362ca6af04 100644
--- a/addons/mass_mailing/static/src/scss/mass_mailing.wysiwyg.scss
+++ b/addons/mass_mailing/static/src/scss/mass_mailing.wysiwyg.scss
@@ -8,6 +8,11 @@
     }
 }
 .o_mass_mailing_iframe body {
+    #oe_manipulators {
+        // Sidebar is already in + 1
+        z-index: $o-we-overlay-zindex + 2;
+    }
+
     .modal:not(.o_technical_modal) {
         top: 0 !important;
         // set z-index so customize options visible on dialog.
```

Reply:
Yes, it worked. I have applied the changes.

Reply:
@dhba-odoo Great, thanks! I'm moving the task back to RD Dev green to be tested again since the change is completely different now than what it was when it was tested the first time.